utils_nlp/models/transformers/sequence_classification.py

```diff
@@ -1,2 +1,117 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
\ No newline at end of file
+# Licensed under the MIT License.
+
+import torch
+from pytorch_transformers.modeling_bert import (
+    BERT_PRETRAINED_MODEL_ARCHIVE_MAP,
+    BertForSequenceClassification,
+)
+from pytorch_transformers.modeling_distilbert import (
+    DISTILBERT_PRETRAINED_MODEL_ARCHIVE_MAP,
+    DistilBertForSequenceClassification,
+)
+from pytorch_transformers.modeling_roberta import (
+    ROBERTA_PRETRAINED_MODEL_ARCHIVE_MAP,
+    RobertaForSequenceClassification,
+)
+from pytorch_transformers.modeling_xlnet import (
+    XLNET_PRETRAINED_MODEL_ARCHIVE_MAP,
+    XLNetForSequenceClassification,
+)
+from torch.utils.data import DataLoader, DistributedSampler, RandomSampler, TensorDataset
+
+from utils_nlp.dataset.pytorch import SCDataSet
+from utils_nlp.models.transformers.common import MAX_SEQ_LEN, TOKENIZER_CLASS, fine_tune
+
+MODEL_CLASS = {}
+MODEL_CLASS.update({k: BertForSequenceClassification for k in BERT_PRETRAINED_MODEL_ARCHIVE_MAP})
+MODEL_CLASS.update(
+    {k: RobertaForSequenceClassification for k in ROBERTA_PRETRAINED_MODEL_ARCHIVE_MAP}
+)
+MODEL_CLASS.update({k: XLNetForSequenceClassification for k in XLNET_PRETRAINED_MODEL_ARCHIVE_MAP})
+MODEL_CLASS.update(
+    {k: DistilBertForSequenceClassification for k in DISTILBERT_PRETRAINED_MODEL_ARCHIVE_MAP}
+)
+
+
+def _list_supported_models():
+    return list(MODEL_CLASS)
+
+
+def create_dataset_from_df(df, text_col, label_col):
+    return SCDataSet(df, text_col, label_col)
+
+
+class Processor:
```

Comment:
Sorry I still don't fully appreciate the need for a class here. Why do we need to make tokenizer and custom_tokenize class properties?

Reply:
It's for loading the tokenizer once in the constructor so it can be reused multiple times for different datasets (train and test for ex). It's also for hiding the details of loading the specific tokenizer. We don't really need to use classes at all - it's just neater.